lib/utils/createElement.js

```diff
@@ -83,28 +84,30 @@ export function table(items, callback) {
 }
 
 export function tabMenuItem(options) {
-	const menu = document.querySelector('#header-bottom-left ul.tabmenu');
+	let menu = document.querySelector('#header-bottom-left ul.tabmenu');
+
+	// Build missing menu if on some of the sites that doesn't have one
+	if (!menu && isPageType('search', 'modqueue')) {
+		menu = $('<ul>', { class: 'tabmenu' })
+			.appendTo('#header-bottom-left')
+			.get(0);
+	} else if (!menu) {
+		console.error('Could not find tab menu');
+		return document.createElement('a');
+	}
 
 	const element = $(tabMenuLinkTemplate(options))[0];
 
-	try {
-		menu.appendChild(element);
-	} catch (e) {
-		console.error('Could not find tab menu');
-		return {
-			label: document.createElement('div'),
-			checkbox: document.createElement('input'),
-		};
-	}
+	if (options.prepend) $(menu).prepend(element);
+	else menu.appendChild(element);
 
-	const checkbox = element.querySelector('input');
-	checkbox.addEventListener('change', e => {
-		e.stopPropagation();
-		element.classList.toggle('selected', checkbox.checked);
+	let checked = options.checked;
+	const a = element.querySelector('a');
+	a.addEventListener('click', () => {
+		checked = !checked;
+		element.classList.toggle('selected', checked);
+		a.dispatchEvent(new CustomEvent('change', { detail: checked }));
```

Comment:
Note: confirmed that this works in Firefox, which is very restrictive about what you can use in `e.detail`